autonomy/chain/tx.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2023 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Tx settlement helper."""
+
+import time
+from datetime import datetime
+from typing import Any, Callable, Dict, Optional
+
+from aea.configurations.data_types import PublicId
+from aea.crypto.base import Crypto, LedgerApi
+from requests.exceptions import ConnectionError as RequestsConnectionError
+
+from autonomy.chain.base import registry_contracts
+from autonomy.chain.config import ChainType, ContractConfigs
+from autonomy.chain.exceptions import ChainTimeoutError, RPCError, TxBuildError
+
+
+try:
+    from web3.exceptions import Web3Exception
+except (ModuleNotFoundError, ImportError):
+    Web3Exception = Exception
+
+DEFAULT_ON_CHAIN_INTERACT_TIMEOUT = 60.0
+DEFAULT_ON_CHAIN_INTERACT_RETRIES = 5.0
+DEFAULT_ON_CHAIN_INTERACT_SLEEP = 3.0
+
+ERRORS_TO_RETRY = (
+    "FeeTooLow",
+    "wrong transaction nonce",
+    "INTERNAL_ERROR: nonce too low",
+)
```

Comment:
@jmoreira-valory @0xArdi If one of these errors is found in the exception, the tx will be retried else we'll raise an error